### ISSUE TITLE
hsdp_container_host_instances does not contain private ips

### ISSUE THREAD

**Issue by jobaeurle** [philips-software/terraform-provider-hsdp#180]
The documentation says that the field for the private ips is named `private_ips`

> [private_ips](https://registry.terraform.io/providers/philips-software/hsdp/latest/docs/data-sources/container_host_instances#private_ips) - The list of container host private IPs. This matches up with the ids list index. 

but in terraform there is only the field `private_addresses` to use, and this is always `null`.
Also for small convenience it would be nice to have the possibility to already select a specific host instance by name like for security group details, something like
```
data "hsdp_container_host_instances" "my_instance" {
  name = "my_instance_name"
}
```

**Comment by loafoe** [comment]
@jobaeurle thanks for the bug report and feature request :)

The `private_ips` is now populated correctly. Also a new [hsdp_container_host](https://registry.terraform.io/providers/philips-software/hsdp/latest/docs/data-sources/container_host) data source was added which provides a lot more details on individual CH instances. Example:

```hcl
data "hsdp_container_host" "my_instance" {
  name = "my_instance_name"
}

output "private_ip" {
  value = data.hsdp_container_host.my_instance.private_ip
}

output "ldap_groups" {
  value = data.hsdp_container_host.my_instance.ldap_groups
}
```

You can try it  out in the `v0.30.11` release.

**Comment by jobaeurle** [comment]
Thank you for the quick reaction :)
The new ´hsdp_container_host` is working perfectly, and the `private_ips` is not null anymore, but I get no values but empty strings for every instance?
e.g. (i replaced the real data with ***)

```
data "hsdp_container_host_instances" "all" {
} 

output "container-all" {
    description = "..."
    value = data.hsdp_container_host_instances.all
}

-----
Changes to Outputs:
container-all = {
      + id          = "cartel_instances"
      + ids         = [
          + "***",
          + "***",
        ]
      + names       = [
          + "***",
          + "***",
        ]
      + owners      = [
          + "***",
          + "***",
        ]
      + private_ips = [
          + "",
          + "",
        ]
      + types       = [
          + "",
          + "",
        ]
    }
```

**Comment by loafoe** [comment]
Ouch, it looks like the Cartel API only returns partial details in the `GetAllInstances()`. 

To not make the call super expensive I will only expose the returned fields. The new [hsdp_container_host](https://registry.terraform.io/providers/philips-software/hsdp/latest/docs/data-sources/container_host) data source can be used to get more details. New release incoming..

**Comment by loafoe** [comment]
@jobaeurle should be fixed in `v0.30.12` -- note that the fix removes `private_ips` and `types` as exported attributes. You can do something like this now (although I'm not sure how practical it is):

```hcl
data "hsdp_container_host_instances" "all" {
}

data "hsdp_container_host" "details" {
  for_each = toset(data.hsdp_container_host_instances.all.names)

  name = each.key
}
```

**Comment by jobaeurle** [comment]
Now it works like expected, perfect. For my use case the new `hsdp_container_host` data source already solved my problem, but it was a bit weird that there were no ips for the instances data source, but since it's removed everything is fine I guess :)
Again thank you very much!

Looking at the changes most were pretty straightforward, if I encounter anything else in future (let's hope that won't be the case), would you be happy if I'd create a pull request or should I create an issue again?
